docs/how-to-guides/effective-gno.md

```diff
@@ -0,0 +1,652 @@
+1. It establishes the initial state, specifically, setting up global variables.
+2. It communicates with another realm, for example, to register itself in a registry.
+
+```go
+import "gno.land/r/some/registry"
+
+func init() {
+    registry.Register("myID", myCallback)
+}
+
+func myCallback(a, b string) { /* ... */ }
+```
+
+A common use case could be to set the "admin" as the caller uploading the
+package.
+
+```go
+import (
+    "std"
+    "time"
+)
+
+var (
+    created time.Time
+    admin   std.Address
+    list    []string
+)
+
+func init() {
+    created = time.Now()
+    admin = std.GetOrigCaller()
+    list = append(list, "foo", "bar")
+}
+```
+
+In essence, `init()` in Gno is your go-to function for setting up and
+registering realms. It's a powerful tool that helps keep your realms organized
+and properly configured from the get-go. Acting as a constructor, it sets the
+stage for the rest of your realm's lifetime.
+
+### A little dependency is better than a little copying
+
+In Go, there's a well-known saying by Rob Pike: ["A little copying is better
+than a little dependency"](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=568s).
+This philosophy encourages developers to minimize their dependencies and instead
+copy small amounts of code where necessary. While this approach often makes
+sense in Go, it's not always the best strategy in Gno.
+
+In Gno, especially for `p/` packages, another philosophy prevails, one that is
+more akin to the Node/NPM ecosystem. This philosophy encourages creating small
+modules and leveraging multiple dependencies. The main reason for this shift is
+code readability and trust.
+
+A Gno contract is not just its lines of code, but also the imports it uses. And
+importantly, Gno contracts are not just for developers. For the first time, it
+makes sense for users to check out what they are executing too. Code simplicity,
+explicitness, and trustability are paramount.
+
+Another good reason for creating simple, focused libraries is the composability
+of Go and Gno. Essentially, you can think of each `p/` package as a Lego brick
+in an ever-growing collection, giving more power to users. `p/` in Gno is
+basically a way to extend the standard libraries in a community-driven manner.
+
+Unlike other compiled languages where dependencies are not always well-known and
+clear metrics are lacking, Gno allows for a reputation system not only for the
+called contracts, but also for the dependencies.
+
+For example, you might choose to use well-crafted `p/` packages that have been
+reviewed, audited, and have billions of transactions under their belt, boasting
+super high stability. This approach can make your code smaller and more
+reliable.
+
+In other platforms, an audit usually involves auditing everything, including the
+dependencies. However, in Gno, we can expect that over time, contracts will
+become smaller, more powerful, and partially audited by default, thanks to this
+enforced open-source system.
+
```

Comment:
I think a passage on how we address supply chain attacks is important here. These are much harder to do in Gno, where packages are immutable.

```suggestion

One key difference between the Go and Gno ecosystem is the trust assumption when
adding a new dependency. Dependency code always needs to be vetted, [regardless
of what programming language or ecosystem you're using][sc-attack]. However, in
Gno, you can have the certainty that the author of a package cannot overwrite an
existing, published contract; as that is simply disallowed by the blockchain. In
other words, using existing and widely-used packages reinforces your security
rather than harming it.
[sc-attack]: https://en.wikipedia.org/wiki/Supply_chain_attack

```